controllers/index.js

```diff
@@ -1,9 +1,21 @@
 var getWeather = require('../services/getWeather');
 var Controller = {};
 var Async = require('async');
+var geoip = require('geoip-lite');
 
 Controller.index = function(req, response) {
-  var params = req.query;
+  var params = getParams();
+
+  function getParams() {
+    if(req.query.lat !== undefined && req.query.lon !== undefined) {
+      return req.query;
+    } else {
+        var ip  = '67.177.208.157'; // just a static IP for now
+        var geo = geoip.lookup(ip)['ll'];
```

Comment:
Is the geoip lookup super-fast? That is, no concerns about needing to wait for it to complete?